### DISCUSSION TITLE
Is it possible to define LaTeX-style macros/commands?

### DISCUSSION THREAD

**Discussion by jmhimara** [quarto-dev/quarto-cli#5526] [Q&A] [answered]
What I mean is, define a command that upon rendering will be replaced by some other markdown or html text, as the `newcommand` does in latex. 

For example I would like to define `\macro = [long description....](long url)`, which enables me to type `\macro` instead of the longer text. 

I know you can use `newcommand` for mathjax rendering, but what I'm asking is more general. Can it be used for any type of text, including non-latex.

**Comment by baptiste** [comment, marked as answer]
Two things come to mind:

- [shortcodes](https://quarto.org/docs/extensions/shortcodes.html)
- using code chunks or inline code, e.g.
````
---
title: "URL"
engine: knitr
---


`r my_url = r"([long description....](https://www.omfgdogs.com/)"`

Lorem ipsum dolor sit amet, `r my_url`.

````

**Reply by ute** [reply]
nice hack :-)

**Comment by cscheid** [comment]
In addition to shortcodes, Lua filters are a great use case for this. For example, if you have markdown like this:

```
Here is [my special text]{.my-span} that will render differently.
```

Then your filter can say

```
Span = function(span)
   if span.classes:includes("my-span") then
      return pandoc.Strong(span)
   end
end
```

This simple filter will make "my special text" bold. But because you're in a full programming language, you can do pretty much anything you want.

**Comment by ute** [comment]
Filters is the way to go :-) I have just written a small quarto extension, <s>[**simple-abbreviations**](https://github.com/ute/simple-abbreviations)</s> [**search-replace**](https://github.com/ute/search-replace), that allows you to define text macros in the yaml header. They can also replace urls and include latex formulae and emphasis/bold markdown.

For a url, say Quartos documentation, you would need to add a key to your yaml:
```yaml
search-replace:
  +myqurl: https://quarto.org/docs/extensions/filters.html
```
Then you can use it in your md text, e.g.

```
Read more about creating filters in [Quarto's documentation](+myqurl)
```

**Reply by cscheid** [reply]
This is a cool filter, thanks for sharing!

**Reply by ute** [reply]
Thank you :-)

**Reply by mcanouil** [reply]
@ute Did you submit to appear in Quarto extensions listing? (https://quarto.org/docs/extensions/)

**Reply by ute** [reply]
@mcanouil Thank you for the suggestion, I will certainly do :-) ! 

I have renamed the extension to "search-replace" and made a few improvements.
